src/context/modal-context.tsx

```diff
@@ -67,7 +74,14 @@ const ModalProvider: FC<ModalProviderProps> = ({ children }) => {
       {children}
       {modal && (
         <PopupDialog
-          closeModal={closeModal}
+          closeModal={
```

Comment:
Can create it more readability. 
Implement separate function and just pass it as a prop
`closeModal={handleCloseHandler}`
```ts
const handleCloseHandler = () => {
  customCloseModal ? customCloseModal() : null
  closeModal()
}
```